bin/main

```diff
@@ -1,3 +1,125 @@
 #!/usr/bin/env ruby
+# rubocop:disable all
+# frozen_string_literal: true
 
-puts 'hello world!'
+def welcome
+  puts 'Welcome to the Tic Tac Toe game!'
+  puts 'do you want to proceed? (y/n)'
+  user_answer = gets.chomp.downcase
+  if user_answer.match(/y/)
+    true
+  elsif user_answer.match(/n/)
+    puts 'bye'
+    false
+  else
+    puts 'wrong answer'
+    welcome
+  end
+end
+
+def instructions
+  puts "Instructions:\n 1)The game is played on a grid that's 3 squares by 3 squares.\n
+   2)Write a letter from A to C and a number from 1 to 3 to select the row and the column respectively that you want to play on.\n
+   3)Player 1 is X, player 2 is O.\n
+   4)Once a player gets 3 marks in a row (up, down, across, or diagonally) is the winner.\n
+   5)If all 9 squares are full, the game is over \n \n"
+end
+
+def players_setting
+  puts "Okay master! May I have first player's name?"
+  user_answer = gets.chomp
+  # some validation
+  $player_one = user_answer
+  puts "cool! #{$player_one}, what about second player's name?"
+  user_answer = gets.chomp
+  # some validation
+  $player_two = user_answer
+  puts "cool! #{$player_two}, let's start to play "
+end
+
+def user_choice
+  whos_turn?
+  puts 'Write a letter from A to C and a number from 1 to 3'
+  payer_call = gets.chomp.upcase
+  round
+  if payer_call.include? 'A'
+    if payer_call.include? '1'
+      $board[0][0] = $letter
+    elsif payer_call.include? '2'
+      $board[0][1] = $letter
+    elsif payer_call.include? '3'
+      $board[0][2] = $letter
+    end
+  elsif payer_call.include? 'B'
+    if payer_call.include? '1'
+      $board[1][0] = $letter
+    elsif payer_call.include? '2'
+      $board[1][1] = $letter
+    elsif payer_call.include? '3'
+      $board[1][2] = $letter
+    end
+  elsif payer_call.include? 'C'
+    if payer_call.include? '1'
+      $board[2][0] = $letter
+    elsif payer_call.include? '2'
+      $board[2][1] = $letter
+    elsif payer_call.include? '3'
+      $board[2][2] = $letter
+    end
+  else
+    puts 'Please write a letter from A to C followed by a number from 1 to 3'
+  end
+  board_matrix
+end
```

Comment:
Instead of working with the square board, it's better to work with an array of length 9.
0 = 0, 0
1 = 0, 1
2 = 0, 2
3 = 1, 0
4 = 1, 1
etc...

from an `index` between `[0, 9]`, `row = index / 3` and `col = index % 3`